lib/adagrams.rb

```diff
@@ -0,0 +1,90 @@
+require "awesome_print"
+
+def draw_letters
+  avail_letters = %w[A A A A A A A A A B B C C D D D D E E E E E E E E E E E E F F G G G H H I I I I I I I I I J K L L L L M M N N N N N N O O O O O O O O P P Q R R R R R R S S S S T T T T T T U U U U V V W W X Y Y Z]
+  avail_letters.sample(10)
+end
+
+sample = draw_letters
+# print sample
+
+def uses_available_letters?(input, letters_in_hand)
+  letters_in_hand = letters_in_hand.clone
+  input = input.upcase
+  input.each_char.all? do |char|
+    if letters_in_hand.include?(char)
+      letters_in_hand.delete_at(letters_in_hand.index(char))
+      true
+    else
+      false
+    end
+  end
```

Comment:
VERY interesting and good job of implementing your own `.all?` logic! I love it! In general, I think that understanding the implicit return of this method might be confusing, so I might recommend rewriting this code to be something like, and take advantage of variables storing what the `.all?` block turns into, and using the explicit return

I have an example of maybe what I would do:

```ruby
is_valid = input.each_char.all? do |char|
    if letters_in_hand.include?(char)
      letters_in_hand.delete_at(letters_in_hand.index(char))
      true
    else
      false
    end
  end
return is_valid
```